skorch/callbacks/logging.py

```diff
@@ -116,97 +117,171 @@ class NeptuneLogger(Callback):
+        self._metric_logger['model/summary'] = NeptuneLogger._model_summary_file(net.module_)
+
+        self._metric_logger['config/optimizer'] = NeptuneLogger._get_obj_name(net.optimizer_)
+        self._metric_logger['config/criterion'] = NeptuneLogger._get_obj_name(net.criterion_)
+        self._metric_logger['config/lr'] = net.lr
+        self._metric_logger['config/epochs'] = net.max_epochs
+        self._metric_logger['config/batch_size'] = net.batch_size
+        self._metric_logger['config/device'] = net.device
+
     def on_batch_end(self, net, **kwargs):
         if self.log_on_batch_end:
             batch_logs = net.history[-1]['batches'][-1]
 
             for key in filter_log_keys(batch_logs.keys(), self.keys_ignored_):
-                self.experiment.log_metric(key, batch_logs[key])
-
-        self.first_batch_ = False
+                self._log_metric(key, batch_logs, batch=True)
 
     def on_epoch_end(self, net, **kwargs):
         """Automatically log values from the last history step."""
-        history = net.history
-        epoch_logs = history[-1]
-        epoch = epoch_logs['epoch']
+        epoch_logs = net.history[-1]
 
         for key in filter_log_keys(epoch_logs.keys(), self.keys_ignored_):
-            self.experiment.log_metric(key, x=epoch, y=epoch_logs[key])
+            self._log_metric(key, epoch_logs, batch=False)
 
     def on_train_end(self, net, **kwargs):
+        self._upload_checkpoint(net)
+
+        try:
+            self._metric_logger['train/epoch/event_lr'].log(net.history[:, 'event_lr'])
+        except KeyError:
+            pass
+
         if self.close_after_train:
-            self.experiment.stop()
+            self.run.stop()
 
+    def _log_metric(self, name, logs, batch):
+        name_parts = name.split('_', maxsplit=1)
+
+        if len(name_parts) == 1:
+            if name == 'dur':
+                key = 'epoch_duration'
+            else:
+                key = name
+            self._metric_logger[key].log(logs[name])
+        else:
+            kind, key = name_parts
+            if kind == 'valid':
+                kind = 'validation'
+
+            if batch:
+                granularity = 'batch'
+            else:
+                granularity = 'epoch'
+
+            # for example:     train /   epoch   / loss
+            self._metric_logger[kind][granularity][key].log(logs[name])
+
+    @staticmethod
+    def _model_summary_file(model):
+        try:
+            # neptune-client=0.9.0+ package structure
+            from neptune.new.types import File
+        except ImportError:
+            # neptune-client>=1.0.0 package structure
+            from neptune.types import File
+
+        return File.from_content(str(model), extension='txt')
+
+    @staticmethod
+    def _get_checkpoint_dirname(net):
+        for callback in net.callbacks:
+            if isinstance(callback, skorch.callbacks.training.Checkpoint):
```

Comment:
This is a bit of a crude way of determining the checkpoint file name. E.g., there can be more than one `Checkpoint` or users could use `TrainEndCheckpoint` or some other custom class. I think I would prefer for this to be more explicit, e.g. the user could pass the checkpoint dirname as an argument. WDYT?

Reply:
@BenjaminBossan thanks, I wasn't aware of that. I'll remove this functionality from the callback and instead document how tracking the checkpoints can be achieved by just providing the path to the tracked files to `Run` object.